image/cas/layout/dir.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2016 The Linux Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package layout
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/opencontainers/image-tools/image/cas"
+	"github.com/opencontainers/image-tools/image/layout"
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+)
+
+// DirEngine is a cas.Engine backed by a directory.
+type DirEngine struct {
+	path string
+	temp string
+}
+
+// NewDirEngine returns a new DirEngine.
+func NewDirEngine(ctx context.Context, path string) (eng cas.Engine, err error) {
+	engine := &DirEngine{
+		path: path,
+	}
+
+	err = layout.CheckDirVersion(ctx, engine.path)
+	if err != nil {
+		return nil, err
+	}
+
+	engine.temp = filepath.Join(path, "tmp")
```

Comment:
This isn't deleted when I run `oci-cas`. Should there be a defer statement somewhere (not here obviously but somewhere else)? Maybe add a cleanup method for `DirEngine` and defer it in the `oci-cas` caller?